Switch to Zig package manager

silversquirl opened the pull request:
Related: https://github.com/godot-zig/godot-zig/pull/10

shiena commented:
@silversquirl 
Thank you for switching to package manager, it would be easier to use if we could also set precision and godot_path.

```diff
diff --git a/build.zig b/build.zig
index c1ec230..a103c77 100755
--- a/build.zig
+++ b/build.zig
@@ -4,9 +4,14 @@ pub fn build(b: *std.Build) void {
     const target = b.standardTargetOptions(.{});
     const optimize = b.standardOptimizeOption(.{});
 
+    const precision = b.option([]const u8, "precision", "Floating point precision, either `float` or `double` [default: `float`]") orelse "float";
+    const godot_path = b.option([]const u8, "godot", "Path to Godot engine binary [default: `godot`]") orelse "godot";
+
     const godot_dep = b.dependency("godot", .{
         .target = target,
         .optimize = optimize,
+        .precision = precision,
+        .godot = godot_path,
     });
 
     const lib = b.addSharedLibrary(.{
@@ -20,7 +25,7 @@ pub fn build(b: *std.Build) void {
     b.installArtifact(lib);
 
     const run_cmd = b.addSystemCommand(&.{
-        "godot", "--path", "./project",
+        godot_path, "--path", "./project",
     });
     run_cmd.step.dependOn(b.getInstallStep());
     const run_step = b.step("run", "run with Godot");
```

silversquirl commented:
@shiena I've added back the godot path option, however exposing `precision` to the user doesn't make sense as it's something that should be decided on and stuck to, not something that should be configurable when building. I've added an example of how to provide it, and a comment clarifying that.
